tests/driver_lsm6dsl/main.c

```diff
@@ -12,28 +12,590 @@
+        printf("Usage: %s <d4d_en> <sixd_ths> <low_pass_on_6d>\n", argv[0]);
+        return 1;
+    }
+
+    lsm6dsl_6d4d_params_t param_6d4d = { .d4d_en = atoi(argv[1]),
+                                         .sixd_ths = atoi(argv[2]),
+                                         .low_pass_on_6d = atoi(argv[3])};
+    if (lsm6dsl_acc_configure_6D4D_alert(&dev, &param_6d4d) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+    puts("[SUCCESS]\n");
+    return 0;
+}
+
+static int _lsm6dsl_cfg_tap_int_handler(int argc, char** argv)
+{
+    if (argc < 6 ) {
+        printf("Usage: %s <tap_ths> <shock> <quiet> <duration> <dbl_tap>\n",
+                argv[0]);
+        return 1;
+    }
+
+    lsm6dsl_tap_params_t tap_param = { .tap_ths = atoi(argv[1]),
+                                         .shock = atoi(argv[2]),
+                                         .quiet = atoi(argv[3]),
+                                         .duration = atoi(argv[4]),
+                                         .dbl_tap = atoi(argv[5])};
+    if (lsm6dsl_acc_configure_tap_alert(&dev, &tap_param) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+    puts("[SUCCESS]\n");
+    return 0;
+}
+
+static int _lsm6dsl_cfg_ofs_xyz_handler(int argc, char** argv)
+{
+    if (argc < 4 ) {
+        printf("Usage: %s <X-offset> <Y-offset> <Z-offset> \n", argv[0]);
+        return 1;
+    }
+
+    if (lsm6dsl_acc_set_x_offset(&dev, atoi(argv[1])) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+
+    if (lsm6dsl_acc_set_y_offset(&dev, atoi(argv[1])) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+
+    if (lsm6dsl_acc_set_z_offset(&dev, atoi(argv[1])) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+    return 0;
+}
+
+static int _lsm6dsl_cfg_ofs_x_handler(int argc, char** argv)
+{
+    if (argc < 2 ) {
+        printf("Usage: %s <X-offset>\n", argv[0]);
+        return 1;
+    }
+
+    if (lsm6dsl_acc_set_x_offset(&dev, atoi(argv[1])) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+    return 0;
+}
+
+static int _lsm6dsl_cfg_ofs_y_handler(int argc, char** argv)
+{
+    if (argc < 2 ) {
+        printf("Usage: %s <Y-offset>\n", argv[0]);
+        return 1;
+    }
+
+    if (lsm6dsl_acc_set_y_offset(&dev, atoi(argv[1])) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+    return 0;
+}
+
+static int _lsm6dsl_cfg_ofs_z_handler(int argc, char** argv)
+{
+    if (argc < 2 ) {
+        printf("Usage: %s <Z-offset>\n", argv[0]);
+        return 1;
+    }
+
+    if (lsm6dsl_acc_set_z_offset(&dev, atoi(argv[1])) != LSM6DSL_OK) {
+        puts("[ERROR]");
+        return 1;
+    }
+    return 0;
+}
+
+static void _usage_configure(void)
+{
+    printf("Usage: configure <wkup_int | sleep_int | 6d4d_int | ff_int | " \
+                "tap_int | ofs_xyz | ofs_x | ofs_y | ofs_z>\n");
+    puts(" * wkup_int & sleep_int configuration layout:");
```

Comment:
Without looking at the code, this usage function is pretty useless. The configuration layout are copy-pasta from the header struct.